nativelib/src/main/scala/scala/scalanative/posix/sys/types.scala

```diff
@@ -0,0 +1,36 @@
+package scala.scalanative
+package posix.sys
+
+import scala.scalanative.native.{CInt, CStruct0, Ptr, extern}
+
+@extern
+object types {
+
+  type pthread_attr_t = Ptr[CStruct0]
+
+  type pthread_barrier_t = Ptr[CStruct0]
+
+  type pthread_barrierattr_t = Ptr[CStruct0]
+
+  type pthread_cond_t = Ptr[CStruct0]
+
+  type pthread_condattr_t = Ptr[CStruct0]
+
+  type pthread_key_t = Ptr[CStruct0]
+
+  type pthread_mutex_t = Ptr[CStruct0]
+
+  type pthread_mutexattr_t = Ptr[CStruct0]
+
+  type pthread_once_t = Ptr[CStruct0]
+
+  type pthread_rwlock_t = Ptr[CStruct0]
+
+  type pthread_rwlockattr_t = Ptr[CStruct0]
+
+  type pthread_spinlock_t = Ptr[CStruct0]
+
+  type pthread_t = Ptr[CStruct0]
```

Comment:
On Linux we'd end up with a nice `type pthread_t = ULong`.
MacOS complicates things:
```c
// /usr/include/sys/_pthread/_pthread_t.h
typedef __darwin_pthread_t pthread_t;
// /usr/include/sys/_pthread/_pthread_types.h
struct _opaque_pthread_t {
	long __sig;
	struct __darwin_pthread_handler_rec  *__cleanup_stack;
	char __opaque[__PTHREAD_SIZE__];
};
typedef struct _opaque_pthread_t *__darwin_pthread_t;
```
That `_opaque_pthread_t` isn't so opaque after all ;-)

I wonder if it's okay to coerce the struct pointer to a `ULong` on Mac.